treeherder/model/models.py

```diff
@@ -697,6 +698,7 @@ class BugJobMap(models.Model):
     bug_id = models.PositiveIntegerField(db_index=True)
     created = models.DateTimeField(default=timezone.now)
     user = models.ForeignKey(User, on_delete=models.CASCADE, null=True)  # null if autoclassified
+    bugscache = models.ForeignKey(Bugscache, on_delete=models.CASCADE)
```

Comment:
It seems like the foreign key might be best on the existing `bug_id`?

Reply:
Yeah, I wondered about that :) If that's changed would I have to add bug_id as a primaryKeyRelatedField on the BugJobMapSerializer and then possibly in Bugscache create an index on the id, for queries by bug_id (via BugJobMap)?

Reply:
@camd, thoughts?

Reply:
@camd I also think I need to add a foreign key on the option_collection_hash field in the Job class, otherwise I need to make two queries for my bug details endpoint.

Reply:
I have to admit, I'm still trying to page-in (as Ed would say) these specifics about DRF.  But if all you want is the bug number, then the ``primaryKeyRelatedField`` sounds right.  If you want other parts of the ``Buscache`` data, then I think you'd want to specify the ``Bugscache`` serializer there and it will put the detail data of the bug nested in your response:  http://www.django-rest-framework.org/api-guide/relations/#nested-relationships

Reply:
Yeah, it seems like Bugscache.id should be a primary key (non auto-increment, obviously) and that will give it an index as well.

Reply:
It may work to add a foreign key for ``option_collection_hash``.  Though what I'd be tempted to try would be to just download all the OptionCollections and store the object keyed off the ``option_collection_hash``.  Then you can reference that in the UI of your code.  It seems like that would save some download payload size as well as adding another index to the job table.

Reply:
So what I'd need to do is iterate through each option_collection_hash and match it to the option__name (option is a foreign key on OptionCollection) and then somehow match that back to the appropriate job_id. Two queries and then mashing data together.

Reply:
Oh, I was thinking of making an object map of {valueYouHave: valueYouWant} and making that global, or at least available to your object.  Then in the render code for the job, just do <span>{options[job.option_name]}</span>  

Or maybe I'm missing what you're trying to do there.  :)

Reply:
I don't think that would work because each bug may have more than one option_collection_hash associated with it, unless you're still querying OptionCollection every time to get the map data and then matching it in the UI. But then it seems it'd make more sense to do that in the server.

Reply:
Yeah, exactly.  :)